test/python/transpiler/test_star_prerouting.py

```diff
@@ -0,0 +1,495 @@
+        qc.h(9)
+        qc.cx(9, range(8, 4, -1))
+
+        pm = PassManager()
+        pm.append(StarPreRouting())
+        pm.append(FinalizeLayouts())
+        result = pm.run(qc)
+
+        self.assertEqual(Operator.from_circuit(result), Operator(qc))
+
+    def test_linear_ghz_no_change(self):
+        qc = QuantumCircuit(6)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.cx(1, 2)
+        qc.cx(2, 3)
+        qc.cx(3, 4)
+        qc.cx(4, 5)
+
+        pm = PassManager()
+        pm.append(StarPreRouting())
+        pm.append(FinalizeLayouts())
+
+        result = pm.run(qc)
+
+        self.assertEqual(Operator.from_circuit(result), Operator(qc))
+
+    def test_no_star(self):
+        qc = QuantumCircuit(6)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.cx(3, 2)
+        qc.cx(0, 3)
+        qc.cx(0, 4)
+        qc.cx(1, 4)
+        qc.cx(2, 1)
+
+        pm = PassManager()
+        pm.append(StarPreRouting())
+        pm.append(FinalizeLayouts())
+        result = pm.run(qc)
+
+        self.assertTrue(Operator.from_circuit(result).equiv(qc))
+
+    def test_10q_bv(self):
+        num_qubits = 10
+        qc = QuantumCircuit(num_qubits, num_qubits - 1)
+        qc.x(num_qubits - 1)
+        qc.h(qc.qubits)
+        for i in range(num_qubits - 1):
+            qc.cx(i, num_qubits - 1)
+        qc.barrier()
+        qc.h(qc.qubits[:-1])
+        for i in range(num_qubits - 1):
+            qc.measure(i, i)
+        result = StarPreRouting()(qc)
+
+        expected = QuantumCircuit(num_qubits, num_qubits - 1)
+        expected.h(0)
+        expected.h(1)
+        expected.h(2)
+        expected.h(3)
+        expected.h(4)
+        expected.h(5)
+        expected.h(6)
+        expected.h(7)
+        expected.h(8)
+        expected.x(9)
+        expected.h(9)
+        expected.cx(0, 9)
+        expected.cx(1, 9)
+        expected.swap(1, 9)
+        expected.cx(2, 1)
+        expected.swap(2, 1)
+        expected.cx(3, 2)
+        expected.swap(3, 2)
+        expected.cx(4, 3)
+        expected.swap(4, 3)
+        expected.cx(5, 4)
+        expected.swap(5, 4)
+        expected.cx(6, 5)
+        expected.swap(6, 5)
+        expected.cx(7, 6)
+        expected.swap(7, 6)
+        expected.cx(8, 7)
+        expected.barrier()
+        expected.h(0)
+        expected.h(1)
+        expected.h(2)
+        expected.h(3)
+        expected.h(4)
+        expected.h(5)
+        expected.h(6)
+        expected.h(8)
+        expected.h(9)
+        expected.measure(0, 0)
+        expected.measure(9, 1)
+        expected.measure(1, 2)
+        expected.measure(2, 3)
+        expected.measure(3, 4)
+        expected.measure(4, 5)
+        expected.measure(5, 6)
+        expected.measure(6, 7)
+        expected.measure(8, 8)
+        self.assertEqual(result, expected)
+
+    # Skip level 3 because of unitary synth introducing non-clifford gates
+    @unittest.skipUnless(HAS_AER, "Aer required for clifford simulation")
+    @ddt.data(0, 1, 2)
```

Comment:
We might need to exclude 2 here too since #12149 we're running unitary synthesis in opt level 2 too. I'll wait to see what CI says though.

Reply:
Yeah sounds good, let's see what CI tells us. :-)

Reply:
I ran CI locally to reduce the pressure on our merge-pipeline and noticed test fails so I excluded level 2, too.